clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,313 @@
+import { UpGradeClientEnums, UpGradeClientInterfaces, UpGradeClientRequests } from '../types';
+import { DefaultHttpClient } from '../DefaultHttpClient/DefaultHttpClient';
+import {
+  CaliperEnvelope,
+  IExperimentAssignmentv5,
+  IFeatureFlag,
+  IFlagVariation,
+  IGroupMetric,
+  ILogInput,
+  ISingleMetric,
+} from 'upgrade_types';
+import { DataService } from 'DataService/DataService';
+import { IApiServiceRequestParams, IEndpoints } from './ApiService.types';
+import { IMarkDecisionPointParams } from 'UpGradeClient/UpGradeClient.types';
+import * as uuid from 'uuid';
+
+// this variable is used by webpack to replace the value of USE_CUSTOM_HTTP_CLIENT with true or false to create two different builds
+declare const USE_CUSTOM_HTTP_CLIENT: boolean;
+declare const IS_BROWSER: boolean;
+
+export default class ApiService {
+  private context: string;
+  private hostUrl: string;
+  private userId: string;
+  private token: string;
+  private apiVersion: string;
+  private clientSessionId: string;
+  private httpClient: UpGradeClientInterfaces.IHttpClientWrapper;
+  private api: IEndpoints;
+
+  constructor(config: UpGradeClientInterfaces.IConfig, private dataService: DataService) {
+    this.context = config.context;
+    this.hostUrl = config.hostURL;
+    this.token = config.token;
+    this.clientSessionId = config.clientSessionId;
+    this.userId = config.userId;
+    this.apiVersion = config.apiVersion;
+    this.api = {
+      init: `${this.hostUrl}/api/${this.apiVersion}/init`,
+      getAllExperimentConditions: `${this.hostUrl}/api/${this.apiVersion}/assign`,
+      markDecisionPoint: `${this.hostUrl}/api/${this.apiVersion}/mark`,
+      setGroupMemberShip: `${this.hostUrl}/api/${this.apiVersion}/groupmembership`,
+      setWorkingGroup: `${this.hostUrl}/api/${this.apiVersion}/workinggroup`,
+      failedExperimentPoint: `${this.hostUrl}/api/${this.apiVersion}/failed`,
+      getAllFeatureFlag: `${this.hostUrl}/api/${this.apiVersion}/featureflag`,
+      log: `${this.hostUrl}/api/${this.apiVersion}/log`,
+      logCaliper: `${this.hostUrl}/api/${this.apiVersion}/log/caliper`,
+      altUserIds: `${this.hostUrl}/api/${this.apiVersion}/useraliases`,
+      addMetrics: `${this.hostUrl}/api/${this.apiVersion}/metric`,
+    };
+    this.httpClient = this.setHttpClient(config.httpClient);
+  }
+
+  private setHttpClient(httpClient: UpGradeClientInterfaces.IHttpClientWrapper) {
```

Comment:
Here it seems like you could import a non-lite build (maybe by accident), and then supply an http client that would be silently ignored in favor of the default one. Probably not a huge issue, but maybe worth documenting?

Reply:
I think that may be true, that would be simple enough to catch an throw a proper error, i can add that.

Reply:
updated and removed console.log